src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -716,18 +722,16 @@ protected Object readResolve() {
             annotations = new ArrayList<>();
         }
 
-        if (yamls == null) {
-            yamls = new ArrayList<>();
-        }
-
-        if (yaml != null) {
-            yamls.add(yaml);
-            yaml = null;
-        }
-
-        // JENKINS-57116 remove empty items from yamls
-        if (!yamls.isEmpty() && StringUtils.isBlank(yamls.get(0))) {
+        // Sanitize empty values
+        yaml = Util.fixEmpty(yaml);
+        if (yamls != null) {
+            // JENKINS-57116 Sanitize empty values
             setYamls(yamls);
+            // Migration from storage in yamls field
+            if (!yamls.isEmpty()) {
+                yaml = yamls.get(0);
```

Comment:
Should this not be verifying that `yamls.size() == 1`?

Reply:
Yes, could even be asserted I guess (unless someone tampered manually with the xml)